AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/GetKeys.dfy

```diff
@@ -440,7 +435,35 @@ module GetKeys {
             branchKeyMaterials := branchKeyMaterials
           ));
     } else if (branchKeyItemFromStorage.EncryptionContext[Structure.HIERARCHY_VERSION] == Structure.HIERARCHY_VERSION_2) {
-
+      // branchKeyItemFromStorage.EncryptionContext comes from storage is not the actual EC.
+      // branchKeyItemFromStorage.EncryptionContext contains all the items in the dynamodb table and table name.
+      var hv2EC := HierarchicalVersionUtils.GetHV2EC(branchKeyItemFromStorage.EncryptionContext);
+      var hv2BranchKey := Types.EncryptedHierarchicalKey(
+        Identifier := branchKeyItemFromStorage.Identifier,
+        Type := branchKeyItemFromStorage.Type,
+        CreateTime := branchKeyItemFromStorage.CreateTime,
+        KmsArn := branchKeyItemFromStorage.KmsArn,
+        EncryptionContext := hv2EC,
+        CiphertextBlob := branchKeyItemFromStorage.CiphertextBlob
+      );
+      var branchKey: KMS.DecryptResponse :- KMSKeystoreOperations.DecryptKeyForHV2(
+        hv2BranchKey,
+        kmsConfiguration,
+        grantTokens,
+        kmsClient
+      );
+      var validateResult := HierarchicalVersionUtils.ValidateMdDigest(branchKey.Plaintext.value, branchKeyItemFromStorage);
+      if (validateResult.Failure?) {
+        return Failure(validateResult.error);
+      }
+      var branchKeyMaterials :- Structure.ToBranchKeyMaterials(
+        branchKeyItemFromStorage,
+        branchKey.Plaintext.value[0..Structure.AES_256_LENGTH]
+      );
+      return Success(
+          Types.GetBranchKeyVersionOutput(
+            branchKeyMaterials := branchKeyMaterials
+          ));
```

Comment:
Blocking: This looks a lot like copy-and-paste/code duplication... 
Can we make a helper method in `HierarchicalVersionUtils` that can be used to consolidate this?